lighthouse-core/gather/gatherers/form-elements.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license Copyright 2019 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+
+/* eslint-env browser, node */
+
+/**
+ *  @param {HTMLElement}
+ *  @return {String}
+ */
+/* istanbul ignore next */
+function getParentForm(node) {
+  if (node == undefined){
+    return [undefined, false];
+  }
+  if (node.nodeName == 'BODY'){
+    return [getNodePath(node), false];
+  };
+  if (node.nodeName == 'FORM'){
+    if (node.id && node.id != ""){
```

Comment:
it seems a little odd to me to return either name/id/nodePath... i'm not sure how you'd use the string afterwards without knowing which you're dealing with.

but maybe it makes sense in the audit?

Reply:
It's mainly going to be comparing similarity. so either name or id can work because in the audit that checks if all the form fields have rendered, when checking for each form I will identify and sort it by the identifying string, so name and id can be used interchanably.

for node path, I am still looking into how I can use it for form elements but same concept nodes with matching path strings would be grouped under the same form